test/tests/nodalkernels/constraint_enforcement/tests

```diff
@@ -44,14 +44,57 @@
       type = Exodiff
       exodiff = upper-and-lower-bound_out.e
       detail = 'have no oscillations in the solution, and'
+      cli_args = "Postprocessors/active='active_upper_lm upper_violations active_lower_lm lower_violations'"
     []
     [non_singular]
       input = upper-and-lower-bound.i
       type = RunApp
       absent_out = '[1-9]+[0-9]* of 33 singular values'
       expect_out = '0 of 33 singular values'
       cli_args = "nx=10 num_steps=5 Outputs/exodus=false Outputs/active='' -pc_type svd -pc_svd_monitor"
-      detail = 'have a non-singular matrix'
+      detail = 'have a non-singular matrix, and'
+    []
+    [amg_fail]
+      input = upper-and-lower-bound.i
+      type = RunException
+      expect_err = 'Linear solve did not converge due to DIVERGED_ITS'
+      cli_args = "Outputs/exodus=false Outputs/active='' -pc_type hypre -pc_hypre_type boomeramg -ksp_converged_reason"
+      detail = 'be incompataible with algebraic multigrid'
+    []
```

Comment:
Does not work with AMG because it is saddle-point problem? 

We should be able to implement it without using Lagrange Multiplier as well.

Reply:
Correct. Yea? But then wouldn’t we just be reimplementing what is done with vinewton?